editor/src/clj/editor/game_object_non_editable.clj

```diff
@@ -335,18 +336,16 @@
   (property prototype-desc g/Any ; No protobuf counterpart.
             (dynamic visible (g/constantly false))
             (set (fn [evaluation-context self _old-value new-value]
-                   ;; We use default evaluation-context in queries to ensure the
-                   ;; results are cached. See comment in connect-resource-node.
                    (let [basis (:basis evaluation-context)
                          project (project/get-project basis self)
-                         workspace (project/workspace project)
-                         proj-path->resource (g/node-value workspace :resource-map)]
+                         workspace (project/workspace project evaluation-context)
+                         proj-path->resource (workspace/make-proj-path->resource-fn workspace evaluation-context)]
                      (letfn [(connect-resource [proj-path-or-resource connections]
                                (:tx-data (project/connect-resource-node evaluation-context project proj-path-or-resource self connections)))]
                        (-> (g/set-property self :embedded-component-resource-data->index
                              (prototype-desc->embedded-component-resource-data->index new-value))
                            (into (g/set-property self :referenced-components
-                                   (prototype-desc->referenced-component-resources new-value workspace)))
+                                   (prototype-desc->referenced-component-resources new-value proj-path->resource)))
```

Comment:
Unrelated fix. I discovered we had not fully transitioned to in-transaction `proj-path` resolution introduced in #10208 here.